### PR TITLE
artifact-sync: remove instrument restrictions

### PR DESCRIPTION
**Pull request by brianmajor**

### CODE REVIEW

File: caom2-artifact-sync/src/main/java/ca/nrc/cadc/caom2/artifactsync/ArtifactHarvester.java [hunk 1]
```diff
@@ -152,24 +157,24 @@ public Integer run() throws Exception {
 
             for (ObservationState observationState : observationStates) {
 
-                String observationID = observationState.getURI().getObservationID();
+                DatabaseTransactionManager txnMgr = new DatabaseTransactionManager(observationDAO.getDataSource());
```
Comment (pdowler) [inline caom2-artifact-sync/src/main/java/ca/nrc/cadc/caom2/artifactsync/ArtifactHarvester.java]: You should not create a new one on the loop... see below

File: caom2-artifact-sync/src/main/java/ca/nrc/cadc/caom2/artifactsync/ArtifactHarvester.java [hunk 2]
```diff
@@ -152,24 +157,24 @@ public Integer run() throws Exception {
...
-                // MAST files that begin with a W, X, Y, or Z
-                List<String> acceptedFilePrefixes = Arrays.asList("w", "x", "y", "z");
-                String firstChar = observationID.substring(0, 1).toLowerCase();
-                if (!acceptedFilePrefixes.contains(firstChar)) {
-                    log.debug("Observation " + observationState + " skipped by temporary data volume restriction");
-                } else {
+                try {
 
-                    DatabaseTransactionManager txnMgr = new DatabaseTransactionManager(observationDAO.getDataSource());
+                    txnMgr.startTransaction();
```
Comment (pdowler) [inline caom2-artifact-sync/src/main/java/ca/nrc/cadc/caom2/artifactsync/ArtifactHarvester.java]: should be observationDAO.getTransactionManager().startTransaction();

File: caom2-artifact-sync/src/main/java/ca/nrc/cadc/caom2/artifactsync/ArtifactHarvester.java [hunk 3]
```diff
@@ -216,18 +221,18 @@ public Integer run() throws Exception {
...
-                        txnMgr.rollbackTransaction();
-                        throw t;
-                    } finally {
-                        if (txnMgr.isOpen()) {
-                            txnMgr.commitTransaction();
-                        }
+                    if (!dryrun) {
+                        harvestStateDAO.put(state);
+                        log.debug("Updated artifact harvest state.  Date: " + state.curLastModified);
+                    }
```
Comment (pdowler) [inline caom2-artifact-sync/src/main/java/ca/nrc/cadc/caom2/artifactsync/ArtifactHarvester.java]: You must have commitTransaction() here at the end of the block

File: caom2-artifact-sync/src/main/java/ca/nrc/cadc/caom2/artifactsync/ArtifactHarvester.java [hunk 4]
```diff
@@ -216,18 +221,18 @@ public Integer run() throws Exception {
...
+                } catch (Throwable t) {
+                    txnMgr.rollbackTransaction();
```
Comment (pdowler) [inline caom2-artifact-sync/src/main/java/ca/nrc/cadc/caom2/artifactsync/ArtifactHarvester.java]: These two should also use observationDAO.getTransactionManager()

File: caom2persistence/src/main/java/ca/nrc/cadc/caom2/harvester/state/HarvestSkipURIDAO.java [hunk 5]
```diff
@@ -162,6 +162,15 @@ public void delete(HarvestSkipURI skip) {
         jdbc.update(sql);
     }
 
+    public void delete(String source, String cname) {
```
Comment (pdowler) [inline caom2persistence/src/main/java/ca/nrc/cadc/caom2/harvester/state/HarvestSkipURIDAO.java]: Given the String arguments, this should be implemented with a PreparedStatement; I would not bother now but put a comment so we know about the technical debt. It is internal so it's not like we are exposed to sql injection attacks now but later re-use could be susceptible...

// TODO: re-implement with PreparedStatement

File: caom2-artifact-sync/src/main/java/ca/nrc/cadc/caom2/artifactsync/ArtifactHarvester.java (hunk 3 above)
Comment (brianmajor) [reply caom2-artifact-sync/src/main/java/ca/nrc/cadc/caom2/artifactsync/ArtifactHarvester.java]: The one in the finally will run though...

**Comment by brianmajor** [comment]
The work in this pull request is now bundled with the delete metadata functionality.

**Pull request closed without merging**